lib/chef/resource/lwrp_base.rb

```diff
@@ -35,32 +35,64 @@ class LWRPBase < Resource
 
       # Evaluates the LWRP resource file and instantiates a new Resource class.
       def self.build_from_file(cookbook_name, filename, run_context)
-        resource_class = nil
-        rname = filename_to_qualified_string(cookbook_name, filename)
-
-        class_name = convert_to_class_name(rname)
-        if Resource.const_defined?(class_name, false)
-          Chef::Log.info("#{class_name} light-weight resource is already initialized -- Skipping loading #{filename}!")
-          Chef::Log.debug("Overriding already defined LWRPs is not supported anymore starting with Chef 12.")
-          resource_class = Resource.const_get(class_name)
-        else
-          resource_class = Class.new(self)
+        resource_name = filename_to_qualified_string(cookbook_name, filename)
+
+        node = run_context ? run_context.node : Chef::Node.new
+        existing_class = Chef::ResourceResolver.new(node, resource_name).resolve
+
+        if existing_class
+          Chef::Log.info("Skipping LWRP resource #{filename} from cookbook #{cookbook_name}.  #{existing_class} already defined.")
+          Chef::Log.debug("Overriding resources with LWRPs is not supported anymore starting with Chef 12.")
+          return existing_class
+        end
 
-          Chef::Resource.const_set(class_name, resource_class)
-          resource_class.resource_name = rname
-          resource_class.run_context = run_context
-          resource_class.class_from_file(filename)
+        # We load the class first to give it a chance to set its own name
+        resource_class = Class.new(self)
```

Comment:
So, I'm totally cool just backing this back out, but I added it on what I *think* was a conversation with you, which I clearly didn't understand as well as I thought at the time :)  You were saying you wished LWRPs could be kicked out of Chef::Resource, and I was OK with that, so I wrote this.  Is that accurate?  (Obviously you didn't also mean "and hey, could you make Chef take gobs and gobs of memory reloading LWRPs over and over again while you're at it," that was all me :)

If so, it seems like the right thing to do is to have this thing ask the ResourceResolver if the LWRP already exists instead of `Chef::Resource`, which should have a similar effect to what you are saying.  I'll implement that till your response, so I can have something.

I'll go dig through the source to understand what was triggering the issue, I'm curious.